build/lib/gitpub/gitpub.py

```
@@ -126,23 +140,65 @@ def get_public_repos(self):
         gh_repo_url = self.repos_url
         repos_count = 0  # number of repos whose details are fetched
 
-        # start requests session
-        sess = requests.session()
+
 
         repos = []  # array to store fetched `Repository`
-        page_number = 1  # to account pagination in the api
+        #page_number = 1  to account pagination in the api
+
+        
+        try:
+            url = 'https://api.github.com/graphql'
+            json = { 'query' : '{user(login:"' + self.username + '") {repositories(first:100){edges{ cursor node{ name id stargazers{ totalCount }description url homepageUrl}}totalCount}}}'}
+            headers = {"Authorization": os.environ['OAUTH_KEY']}
```

Same as before

---

Should I hardcode my authentication key then?

---

Yes, it definitely isn't a good way to do the things. 
I didn't think of this before! Have a look [here](https://platform.github.community/t/permit-access-without-oauth-token/2572/15). Seems like there isn't a way to send anonymous requests in GraphQL. Will need to think of a someway to  do this. If nothing turns out, I'll have to create an `OAUTH_KEY` for testing purposes. But anyways, there should be a check for the absence of `OAUTH_KEY`, if it isn't found we should raise an error there.

Anyways, I didn't point this out earlier, but you don't need to change anything in the `build` directory as it will be automatically updated when we uptick GitPub's version.